Clinical-Genomics/scout
# Remove null non-mandatory key/values when updating a case

Fix #3153 --> Avoid saving empty keys (also non specific to that type of case) when updating a case 

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
</details>

**How to test**:
1. Run scout setup demo and check in MongoDB compass that case contains the expected key/values.
2. While on master branch, re-upload the case: `scout --demo load case scout/demo/643594.config.yaml -u` 
3. Make sure that a bunch of non-necessary keys with empty values get saved in the updated case:
 - cnv_report
 - coverage_qc_report
 - gene_fusion_report_research
 - mme_submission
 - multiqc
4. Repeat both steps from this branch and check that these extra key/values don't get saved in case document

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
